From: Simon Richter <sjr@debian.org>
Date: Fri, 29 Aug 2014 16:17:45 +0200
Subject: Debian compliant compiler flags handling

---
 CMakeLists.txt         |  1 -
 backend/CMakeLists.txt | 46 +++++++++++++++++++++++-----------------------
 2 files changed, 23 insertions(+), 24 deletions(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,6 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINA
 
 INCLUDE (FindPkgConfig)
 
-SET(CMAKE_VERBOSE_MAKEFILE "false")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMake/")
 if (NOT LIB_INSTALL_DIR)
   set (LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib")
