Description: fix crash on unsupported hardware

Origin: upstream 24b2adbd958884d5874dc12f456cee9889583967
Author: Zhigang Gong <zhigang.gong@intel.com>
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=88957
diff --git a/src/cl_device_id.c b/src/cl_device_id.c
index 3032a38..0fd4a69 100644
--- a/src/cl_device_id.c
+++ b/src/cl_device_id.c
@@ -395,13 +395,14 @@ brw_gt3_break:
       printf("cl_get_gt_device(): error, unknown device: %x\n", device_id);
   }
 
+  if (ret == NULL)
+    return NULL;
+
   if (!CompilerSupported()) {
-    if (ret != NULL) {
-      ret->compiler_available = CL_FALSE;
-      //ret->linker_available = CL_FALSE;
-      ret->profile = "EMBEDDED_PROFILE";
-      ret->profile_sz = strlen(ret->profile) + 1;
-    }
+    ret->compiler_available = CL_FALSE;
+    //ret->linker_available = CL_FALSE;
+    ret->profile = "EMBEDDED_PROFILE";
+    ret->profile_sz = strlen(ret->profile) + 1;
   }
 
 #ifdef HAS_USERPTR
-- 
cgit v0.10.2

