From: Simon Richter <sjr@debian.org>
Date: Fri, 29 Aug 2014 16:29:44 +0200
Subject: Link against terminfo

---
 backend/src/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/backend/src/CMakeLists.txt
+++ b/backend/src/CMakeLists.txt
@@ -128,6 +128,7 @@ set (GBE_LINK_LIBRARIES
     ${LLVM_SYSTEM_LIBS}
     ${CMAKE_THREAD_LIBS_INIT}
     ${CMAKE_DL_LIBS}
+    tinfo
     )
 
 include_directories (.)
