Description: Allow building tests with Python 3 (no string.atoi)

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Forwarded: https://lists.freedesktop.org/archives/beignet/2016-September/008007.html

--- beignet-1.2.0.orig/utests/utest_generator.py
+++ beignet-1.2.0/utests/utest_generator.py
@@ -248,7 +248,7 @@ which can print more values and informat
   def argvector(self,paraN,index):
     vector=re.findall(r"[0-9]+",self.inputtype[paraN][index])
     if vector:
-      vector=string.atoi(vector[0])
+      vector=int(vector[0])
     else:
       vector=1
     return vector
